### ISSUE TITLE
Why statement coverage is not 100% in this class?

### ISSUE THREAD

**Issue by omerlh** [scoverage/sbt-scoverage#382]
Hello!
I have a lot of small classes, where scoverage says the coverage is not complete - in this case:

![image](https://user-images.githubusercontent.com/6730584/138630884-fe48aa9e-f704-423e-ab28-3ce280e5058b.png)

And I can't figure out why... Is this a bug? Here is the example class (I've redacted the actual code):

![Screen Shot 2021-10-25 at 6 31 15](https://user-images.githubusercontent.com/6730584/138630814-dd69a760-687e-41c8-ad4c-b2a63b4d50ea.png)

Is this a bug? Or am I missing something?

I am using the latest version:
```scala
addSbtPlugin("org.scoverage" % "sbt-scoverage" % "1.9.1")
```

**Comment by ckipp01** [comment]
> I have a lot of small classes, where scoverage says the coverage is not complete - in this case:

So on the page in the first image if you click "Statement List", does it show you the statements that aren't covered? I'm assuming there are 3 looking at the numbers?

**Comment by omerlh** [comment]
This is the statement list: 
![image](https://user-images.githubusercontent.com/6730584/140635651-0db5032d-4e87-4d5e-a969-c06b8856daf9.png)

**Comment by ckipp01** [comment]
🤔 really? That's odd as the first image you showed says there are 35 statements

**Comment by omerlh** [comment]
Yep I know! This is really word. Let me try reproduce it in a clean project I can share publicly...

**Comment by omerlh** [comment]
Also, for some reason the table is incomplete - it does not have the tests or code columns...

**Comment by nigeleke** [comment]
I have another example to add to the "Why statement coverage not 100%"

Code:
```
object Stuff:

  enum Foo:
    case Bar, Baz, Bam

  val wibble =
    (for thing <- Some(42) yield thing).getOrElse(0)

  val wobble =
    (for f <- Stuff.Foo.values yield f).toList
```

Test cases (scalatest):

```
class TestStuff extends AnyWordSpec with Matchers:

  "Stuff" should {

    "use for loop" in {
      val x = Stuff.wibble
      x should be(42)
    }

    "use object and enums" in {
      import Stuff.*
      val x = Stuff.wobble
      x should contain theSameElementsAs (Seq(Foo.Bar, Foo.Baz, Foo.Bam))
    }

  }
```

Test Coverage:

![image](https://user-images.githubusercontent.com/775206/210352151-77881ee8-1037-4913-b547-c0bc2a50c7d2.png)

![image](https://user-images.githubusercontent.com/775206/210352251-849f39d2-5911-4ff8-804f-a3d73e3806a6.png)

**Comment by nigeleke** [comment]
Adding Set(...) example:
![image](https://user-images.githubusercontent.com/775206/210453451-c64b28cc-a68d-4478-bb67-482977c501c7.png)
![image](https://user-images.githubusercontent.com/775206/210453503-a4e4f263-9320-4709-9e32-96b6acd86f8d.png)
![image](https://user-images.githubusercontent.com/775206/210453540-f71c43d7-1963-4a2d-ad69-7cae159744d6.png)
![image](https://user-images.githubusercontent.com/775206/210453577-95c72105-034b-4a69-880c-caa3180342e4.png)
